index.html

```diff
@@ -645,26 +649,41 @@ <h3>The <dfn data-dfn-for="GlobalEventHandlers" data-dfn-type="event">gotpointer
+                <p>These events MUST be of type <code>PointerEvent</code>, and are subject to the additional requirements mentioned in the rest of this section.</p>
+                <section>
+                  <h4>Attributes (other than coordinates)</h4>
+                  <p>For these events, all <code>PointerEvent</code> specific attributes (defined in this spec) other than <code>pointerId</code> and <code>pointerType</code> MUST have their default values. In addition:</p>
+                  <ul>
+                    <li>If the events are generated by a pointing device, their <code>pointerId</code> and <code>pointerType</code> MUST be the same as the PointerEvents that caused these events.</li>
+                    <li>If the events are generated by a non-pointing device (such as voice recognition software or a keyboard interaction), <code>pointerId</code> MUST be <code>-1</code> and <code>pointerType</code> MUST be an empty string.</li>
+                  </ul>
+                </section>
+                <section>
+                  <h4>Event coordinates</h4>
+                  <p>As noted in {{PointerEvent}}, [[[CSSOM-VIEW]]] proposes to redefine the various coordinate properties (<code>screenX</code>, <code>screenY</code>, <code>pageX</code>, <code>pageY</code>, <code>clientX</code>,
+                      <code>clientY</code>, <code>x</code>, <code>y</code>, <code>offsetX</code>, <code>offsetY</code>) as <code>double</code>, to allow for fractional coordinates.
+                      However, this change — when applied only to {{PointerEvent}}, but not to regular {{MouseEvent}} — has proven to lead to web compatibility issues with legacy code
+                      in the case of <code>click</code>, <code>auxclick</code>, and <code>contextmenu</code>. For this reason, user agents that have implemented the proposed
+                      change in [[[CSSOM-VIEW]]] only for {{PointerEvent}} MUST convert the various coordinate properties for the <code>click</code>, <code>auxclick</code>, and <code>contextmenu</code>
+                      to <code>long</code> values (as defined in the original [[[UIEVENTS]]]) using <a data-cite="ECMASCRIPT#sec-math.floor">Math.floor</a> [[ECMASCRIPT]].</p>
+                </section>
+                <section>
+                  <h4>Event dispatch</h4>
+                  <p>If a <code>click</code>, <code>auxclick</code> or <code>contextmenu</code> event |e| is fired in response to a user interaction event |t|, and |t| is a <code>PointerEvent</code> which was dispatched while the corresponding pointer was captured:</p>
+                  <ul>
+                    <li>Event |e| MUST be dispatched to the capturing target of the |t|.  However, if the capturing target is no longer [=connected=] [[DOM]], |e| MUST be dispatched to the document.</li>
```

Comment:
Thanks for noting this.

> Oddly, we do still calculate the new contextmenu target and dispatch the contextmenu.

By "calculate" you meant "hittesting" the `contextmenu` event, right?  That's what I found on both Chrome and Firefox.  On Windows, right-clicking near the lower edge of `#c#` sends the event to `log`!

> TLDR; I feel like we should match this behavior - otherwise capturing to the same element will have different behavior even when the release event is over that element.

Please elaborate the scope of "this behavior": did you mean `contextmenu` event would be hittested even for the capturing case?

Reply:
The simplest least intrusive thing to spec would be that we only define special behavior when there is a capturing target that is still in the dom - otherwise we can back to the standard ui events targeting behavior - which is that we don't generate click and contextmenu is independently targeted.

I haven't tested yet - but I assume with regular pointerevents when the capturing element is removed any subsequent events fall back to the standard non-captured targeting behavior.

Reply:
> The simplest least intrusive thing to spec would be that we only define special behavior when there is a capturing target that is still in the dom - otherwise we can back to the standard ui events targeting behavior - which is that we don't generate click and contextmenu is independently targeted.

Done, ptal.

> I haven't tested yet - but I assume with regular pointerevents when the capturing element is removed any subsequent events fall back to the standard non-captured targeting behavior.

Element removal causes firing of `lostpointercapture` as tested in `pointerevent_lostpointercapture_for_disconnected_node.html`.  This is defined in Sec9.5 but it should be clear that an implicit release could be caused by target removal!  I will handle this in a separate PR.